src/main/java/core/StandardForwardModel.java

```diff
@@ -20,6 +22,22 @@ protected void _next(AbstractGameState currentState, AbstractAction action) {
         } else {
             throw new AssertionError("No action selected by current player");
         }
+        // We then register the action with the top of the stack ... unless the top of the stack is this action
+        // in which case go to the next action
+        // We can't just register with all items in the Stack, as this may represent some complex dependency
+        // For example in Dominion where one can Throne Room a Throne Room, which then Thrones a Smithy
+        if (currentState.actionsInProgress.size() > 0) {
+            IExtendedSequence topOfStack = currentState.actionsInProgress.pop();
+            if (topOfStack != action) {
```

Comment:
Should this not be equals?

Reply:
I put this as a reference level comparison deliberately - but on reflection the edge case I was thinking of is probably too extreme; and your approach is safer.